### PR TITLE
Heretic Buffs, the Rebuffening

### PR DESCRIPTION
**Pull request by SpaceVampire**
## About The Pull Request

This PR gets Heretics back up to the same kind of level that all of our other main antags (Tators and Clings) have been pushed to in order to actually stand a chance in the current balance environment (Health changes, and security changes). Many of the changes in this PR had been made to them already, but were removed with the major rework of the code from upstream.

Buffs their heretic blade to no longer be way worse then basic bone tools, or the standard melee weapon of the other two antags, by giving it some extra damage and AP.
This also lets them make a couple extra copies if the weapon if they end up getting one-clicked disarmed into dropping it, because at that point you are already falling behind and not being able to create a new weapon makes it impossible to come back. I might lower the limit back down if it really becomes an issue, but the current one is very punishing for any mistakes.
Sacrifices now give more points, because of how much harder it is to pull off when we expect them to do more then just teleporting into the room and suddenly ganking people. Less punishing if you try to RP it out.
I have really only seen people pull off two of them per round, max. Most people are still doing less then that.
They also have one extra starting point, partly because I suspect a lot of people are going to want to buy the reroll ritual for the heart, because now you can't just make a second one if your target is busy in dorms.

Expect this thing to change as I get feedback on it, and it gets test-merged hopefully.

## How This Contributes To The Skyrat Roleplay Experience

Having the antag actually be playable, and not a total joke of a threat is good.

## Changelog

:cl:
balance: Heretic sickle-sword now does 8 more damage, and has slight armor pen.
balance: Heretics are now able to have more of their weapons summoned at once.
balance: Heretics start with 1(one) extra free bonus point.
balance: The gift of Mansus becomes ever more temping, as those granted with the power of a Heretic now receive greater strength from sacrificing targets.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/heretic/items/madness_mask.dm [hunk 1]
```diff
@@ -68,7 +68,7 @@
 
 		if(human_in_range.getStaminaLoss() >= 85 && DT_PROB(30, delta_time))
 			human_in_range.emote(pick("giggle", "laugh"))
-			human_in_range.adjustStaminaLoss(10)
+			human_in_range.adjustStaminaLoss(15) //SKYRAT EDIT - Original: 10
```
Comment (RatFromTheJungle) [inline code/modules/antagonists/heretic/items/madness_mask.dm]: this can't afford to be buffed anymore, due to the fact stamina regens extremely slowly now, as compared to during the first PR. (Even just 15 would leave it stamcritting people through walls, regularly)

File: code/modules/antagonists/heretic/items/heretic_blades.dm [hunk 2]
```diff
@@ -12,7 +12,8 @@
 	flags_1 = CONDUCT_1
 	sharpness = SHARP_EDGED
 	w_class = WEIGHT_CLASS_NORMAL
-	force = 17
+	force = 25 //SKYRAT EDIT - ORIGINAL: 17
```
Comment (RatFromTheJungle) [inline code/modules/antagonists/heretic/items/heretic_blades.dm:15]: 25 is a tad strong, 18-20 with 7-10 armor pen would be alot more 'fair'. (20+ is breaking lockers tier, which their knife shouldn't just filter into)

File: code/modules/antagonists/heretic/heretic_antag.dm [hunk 3]
```diff
@@ -30,7 +30,7 @@
 	/// The path our heretic has chosen. Mostly used for flavor.
 	var/heretic_path = PATH_START
 	/// A list of how many knowledge points this heretic CURRENTLY has. Used to research.
-	var/knowledge_points = 1
+	var/knowledge_points = 2 //SKYRAT EDIT - ORIGINAL 1
```
Comment (RatFromTheJungle) [inline code/modules/antagonists/heretic/heretic_antag.dm:33]: their early game is their only weak-ish point now, with never losing the ability to craft more stuff, so a points-gain buff isn't as needed currently.

File: code/modules/antagonists/heretic/knowledge/rust_lore.dm [hunk 4]
```diff
@@ -44,7 +44,7 @@
 		/obj/item/trash = 1,
 	)
 	result_atoms = list(/obj/item/melee/sickly_blade/rust)
-	limit = 2
+	limit = 20 //SKYRAT EDIT - ORIGINAL: 2
```
Comment (RatFromTheJungle) [inline code/modules/antagonists/heretic/knowledge/rust_lore.dm]: 5 at most, so it's not something they can afford to keep throwing away.

File: code/modules/antagonists/heretic/knowledge/ash_lore.dm [hunk 5]
```diff
@@ -44,7 +44,7 @@
 		/obj/item/match = 1,
 	)
 	result_atoms = list(/obj/item/melee/sickly_blade/ash)
-	limit = 2
+	limit = 20 //SKYRAT EDIT - ORIGINAL: 2
```
Comment (RatFromTheJungle) [inline code/modules/antagonists/heretic/knowledge/ash_lore.dm]: 5 at most, so it's not something they can afford to keep throwing away

File: code/modules/antagonists/heretic/knowledge/flesh_lore.dm [hunk 6]
```diff
@@ -49,7 +49,7 @@
 		/obj/effect/decal/cleanable/blood = 1,
 	)
 	result_atoms = list(/obj/item/melee/sickly_blade/flesh)
-	limit = 3 // Bumped up so they can arm up their ghouls too.
+	limit = 30 // Bumped up so they can arm up their ghouls too. //SKYRAT EDIT - ORIGINAL: 3
```
Comment (RatFromTheJungle) [inline code/modules/antagonists/heretic/knowledge/flesh_lore.dm]: honestly, flesh can have the 20, because they need it more then the other two choices.

File: code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm [hunk 7]
```diff
@@ -189,7 +189,7 @@
 		heretic_datum.high_value_sacrifices++
 
 	heretic_datum.total_sacrifices++
-	heretic_datum.knowledge_points += 2
+	heretic_datum.knowledge_points += 4 //SKYRAT EDIT - ORIGINAL: 2
```
Comment (RatFromTheJungle) [inline code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm:192]: sacrifices didn't need a point buff, espiecally with how strong rifts are now.

File: code/modules/antagonists/heretic/knowledge/void_lore.dm [hunk 8]
```diff
@@ -42,7 +42,7 @@
 	)
 	required_atoms = list(/obj/item/knife = 1)
 	result_atoms = list(/obj/item/melee/sickly_blade/void)
-	limit = 2
+	limit = 20 //SKYRAT EDIT - ORIGINAL: 2
```
Comment (RatFromTheJungle) [inline code/modules/antagonists/heretic/knowledge/void_lore.dm]: 5 at most, so it's not something they can afford to keep throwing away.

File: code/modules/antagonists/heretic/items/madness_mask.dm (hunk 1 above)
Comment (SpaceVampire) [reply code/modules/antagonists/heretic/items/madness_mask.dm]: Sure. I'll leave it at the baseline amount for the moment.
I thought that the general HP buff would have effected it, but I'll see how it turns out if people use it.

File: code/modules/antagonists/heretic/heretic_antag.dm (hunk 3 above)
Comment (SpaceVampire) [reply code/modules/antagonists/heretic/heretic_antag.dm:33]: Part of my logic is you can't reroll your heart target for free anymore, and with how Skyrat tends to play a lot of people are really going to feel like they have to buy the ritual for it. I could always just make that one free if you think that is a better option, or if you think that isn't needed at all let me know.

File: code/modules/antagonists/heretic/knowledge/void_lore.dm (hunk 8 above)
Comment (SpaceVampire) [reply code/modules/antagonists/heretic/knowledge/void_lore.dm]: Sure, I'll go with that for now.

File: code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm (hunk 7 above)
Comment (SpaceVampire) [reply code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm:192]: Rifts aren't any stronger unless you spend points to make the book to buff them, which is a risky investment.
I'm willing to lower it down to three points, but I still think they should get a point buff given that you aren't allowed to JUMP people here, in the same way you would be on TG normally.

File: code/modules/antagonists/heretic/knowledge/flesh_lore.dm (hunk 6 above)
Comment (SpaceVampire) [reply code/modules/antagonists/heretic/knowledge/flesh_lore.dm]: Sure, I'll go with that for now.

File: code/modules/antagonists/heretic/knowledge/ash_lore.dm (hunk 5 above)
Comment (SpaceVampire) [reply code/modules/antagonists/heretic/knowledge/ash_lore.dm]: Sure, I'll go with that for now.

File: code/modules/antagonists/heretic/knowledge/rust_lore.dm (hunk 4 above)
Comment (SpaceVampire) [reply code/modules/antagonists/heretic/knowledge/rust_lore.dm]: Sure, I'll go with that for now.

File: code/modules/antagonists/heretic/items/heretic_blades.dm (hunk 2 above)
Comment (SpaceVampire) [reply code/modules/antagonists/heretic/items/heretic_blades.dm:15]: I could lower the AP value on it, but I think the force value is still fine when compared to the other weapons.

Comment (RatFromTheJungle) [reply code/modules/antagonists/heretic/items/heretic_blades.dm:15]: most other weapons aren't mass-producible, as compared to these they can make with a knife and a junk item.

Comment (SpaceVampire) [reply code/modules/antagonists/heretic/items/heretic_blades.dm:15]: I mean, if we are comparing to something like an changeling armblade you aren't able to lose it.
Traitors have martial arts for the same logic, and stuff like the double-energy sword as a stronger melee option.

File: code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm (hunk 7 above)
Comment (SpaceVampire) [reply code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm:192]: After thinking about it, I have decided to make it only one more point for the moment.
I'll see how it feels after I watch a couple shifts play out with the minor buff.

**Comment by hugmeorelse** [comment]
I'm wary of further power creep. I rather have antagonists be on the weaker side than have a station that is constantly in turmoil.

**Comment by SpaceVampire** [comment]
> I'm wary of further power creep. I rather have antagonists be on the weaker side than have a station that is constantly in turmoil.

I understand the concern, which is why I am starting small with this balance patch and working from there. Already went and toned it down from my first plan, before it has even been test-merged. But I really do this is needed, after having watched like six shifts worth of Heretics being unable to do anything of real note. Our security team is just a lot more militarized then TG, and on top of that I think we often have more of them in any given shift. This partly comes down to dynamic config stuff, but I think a lot of people have been turned off from playing antag. We only had like two of them trying to deal with an almost fully staffed team, and that isn't counting stuff like department guards. Both traitors and changelings have got some fairly strong buffs to try and keep up, and I don't think that anyone feels like they are some unstoppable force at the moment.

**Comment by SpookyTheFox** [comment]
Since I've been invited to leave feedback here..
We've had a large number of rounds with heretics dramatically over preforming. The clear problem child and also meta pick, is Rust path, which commits the same sin as space vines. Which is near invincibility while standing on some tiles. Except it's worse than space vines, because unlike space vines, it's very hard to remove rust tiles and very easy to spread them over a large area. This really needs to be addressed. On top of that, multiple, stackable methods of teleportation are making heretics nearly impossible to fight. Stacking different teleportation abilities negates their built in balance attempts. 

One of those teleportation offenders is ash blink, which is available to more than ash path, and another one is blade break. Blade count is altered in this PR. An additional downside of their raised blade count is that it makes heretics difficult if not impossible to disarm, which gets us into a bad spot for role play and meta protections. I would like their blade count reduced from default, not increased. Further, it does not need the extra damage or the armor penetration. That just goes against why we have more health.

In short, drop all the knife stuff both damage and counts. And we need to have a wider look at rust in particular which needs some special restrictions.

I also have some concerns about giving them extra points per sacrifice, because that mechanically encourages them to do something pretty unfair and disruptive, with fewer opportunities to halt them. Also gets this antag into the same situation as prog traitor, who currently is heavily mechanically encouraged to role play as little as possible and speed run points.

**Comment by SpaceVampire** [comment]
> Since I've been invited to leave feedback here.. We've had a large number of rounds with heretics dramatically over preforming. The clear problem child and also meta pick, is Rust path, which commits the same sin as space vines. Which is near invincibility while standing on some tiles. 

What do you count as "dramatically over preforming" exactly? Because I haven't seen a single Heretic asecnd during the many rounds that I have watched and logged during testing, and most of them struggle to even reach two sacrifices. Not trying to be rude, but that is the kind of grand statement where it could be nice to give some kind of example. I also think that "near invincibility" is just a touch much for something that only heals you two damager per tick. Which is about half as much as just drinking basic healing chems, and LESS then drinking 100% pure chems churned out by a factory. 

>Except it's worse than space vines, because unlike space vines, it's very hard to remove rust tiles and very easy to spread them over a large area. This really needs to be addressed. 

I don't really think that "click on them with a welder for half a second" counts as that hard to remove. You know you can just burn rest off with those electric welders stupid quick, right? There is a tool just for it as well, that is also pretty fast. I also don't think it is that easy to spread them, when the main way to do anything other then just a single tile for healing is the spell that has a 40 second cooldown between casts. So that you can rust right where you are fighting, but if you have to run away from that spot it isn't all that easy to get back on the rust again. The only other way is an even more later game spell/summon.

>On top of that, multiple, stackable methods of teleportation are making heretics nearly impossible to fight. Stacking different teleportation abilities negates their built in balance attempts.

I do think that when compared to our the other two main antags, they have the most in the way of escape based powers. But I am not sure where "Stacking different teleportation abilities negates their built in balance attempts." comes from given that you are designed to cross path, that is a pretty basic feature and is actually REQUIRED to unlock the capstone power.

> One of those teleportation offenders is ash blink, which is available to more than ash path, and another one is blade break. Blade count is altered in this PR. An additional downside of their raised blade count is that it makes heretics difficult if not impossible to disarm, which gets us into a bad spot for role play and meta protections. I would like their blade count reduced from default, not increased. Further, it does not need the extra damage or the armor penetration. That just goes against why we have more health.

You would like it if you could only ever have a single blade for any reason? So that when someone hits you with a stun baton one single time, you drop it, and they pick it up it is all over on you for ever having a weapon again? Because the default is two, and there is only one number smaller then that. I'm also not really sure how it makes them impossible to disarm, as the limit is still pretty reasonable. It was just designed to let people make more then a single mistake ever, and to let Flesh path arm their summons which are already relatively weaker due to the health bonus. We buffed the changeling armblade, and we also removed the more damage equals stamina damage part of combat. so it needs the slight damage boost to be an actual weapon. I don't feel like it has been too deadly at all during testing, as it is still worse then stuff you can get from robotics. Which also don't get disarmed from a single hit, once again.
 
> In short, drop all the knife stuff both damage and counts. And we need to have a wider look at rust in particular which needs some special restrictions.

In short, I think I feel like both of those things have been fine during testing. The only thing I really considered doing was giving the "Blade Blade" something of a channel time in to give it a bit more counterplay. Making it that you would have to escape active combat for a moment in order to use it. It is a still a bit of a risky play, because I have seen Heretics teleport directly into the AI update with turrets, and right into the security room next to an angry beepsky during the testing. 

> I also have some concerns about giving them extra points per sacrifice, because that mechanically encourages them to do something pretty unfair and disruptive, with fewer opportunities to halt them. Also gets this antag into the same situation as prog traitor, who currently is heavily mechanically encouraged to role play as little as possible and speed run points.

The whole point is that by giving them a couple of extra points you aren't forced to do extra sacrifices just in the name of getting more points. Given that most people already seem to be having trouble getting more then two of them, I don't think it is a real issue. I also think that more people have been attempting to RP with people because of this, and the fact that the getting sacrificed is a lot more forgiving helps with that as well.

Now that I did my best to address all of your points, I want to make a more general comment. Based on your discord messages about them, it seems like really dislike the fact that they can escape and think they should just be murdered or locked right when they first try something so that there can be a "definitive end" to stuff. Which I feel like is just going to lead to a lot more of people doing nothing at all. I feel like this works when it comes to making them a cool simi-rare antag that isn't going to get locked down before getting a chance to actually do anything. I'm fine with making a couple of changes to try and cut down on the "cat and mouse" nature of it. But in terms of raw power I really think that changelings are stronger, along with traitors most of the time.

**Comment by Vizoplez** [comment]
Hot take:
Rust is nowhere near as dangerous nor meta as void or flesh because both of those can do something rust cannot, survive in space indefinitely, without a suit, and cause massive environmental and mob spam hazards via rituals.

Seriously, nobody does it here but you can *VERY* easily do a lot of damage to the station with nothing but a piece of paper, pen, knife, and a few monkey bodies.
